gtars/src/common/models/region_set.rs

```diff
@@ -84,30 +188,276 @@ impl<'a> IntoIterator for &'a RegionSet {
+                    rest: i.rest.as_deref().unwrap_or("").to_string(),
+                },
+            )))
+        });
+
+        let region_vector = region_vector.filter(|e| e.is_some()).map(|e| e.unwrap());
+
+        let runtime = runtime::Builder::new_multi_thread()
+            .worker_threads(
+                std::thread::available_parallelism()
+                    .map(|c| c.into())
+                    .unwrap_or(1),
+            )
+            .build()
+            .expect("Unable to create thread pool.");
+
+        let mut bb_out = BigBedWrite::create_file(out_path, chrom_sizes.clone())
+            .expect("Failed to create bigBed file.");
+
+        bb_out.options.max_zooms = 8;
+
+        let data = BedParserStreamingIterator::wrap_iter(region_vector.into_iter(), true);
+        bb_out.write(data, runtime)?;
+        Ok(())
+    }
+
+    ///
+    /// Sort bed file based on first 3 columns
+    ///
+    pub fn sort(&mut self) -> () {
+        self.regions
+            .sort_by(|a, b| a.chr.cmp(&b.chr).then_with(|| a.start.cmp(&b.start)));
+    }
+
+    ///
+    /// Get number of regions in RegionSet
+    ///
+    /// Returns:
+    /// number of regions
     pub fn len(&self) -> usize {
         self.regions.len()
     }
+}
+
+impl Display for RegionSet {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "RegionSet with {} regions.", self.len())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn get_test_path(file_name: &str) -> Result<PathBuf, Error> {
+        let file_path: PathBuf = std::env::current_dir()
+            .unwrap()
+            .join("tests/data/regionset")
+            .join(file_name);
+        Ok(file_path)
+    }
+
+    #[test]
+    fn test_open_from_path() {
+        let file_path = get_test_path("dummy.narrowPeak").unwrap();
+        assert!(RegionSet::try_from(file_path.as_path()).is_ok());
+    }
+
+    #[test]
+    fn test_open_from_string() {
+        let file_path = get_test_path("dummy.narrowPeak").unwrap();
+        assert!(RegionSet::try_from(file_path.to_str().unwrap()).is_ok());
+    }
+
+    #[test]
+    fn test_open_bed_gz() {
+        let file_path = get_test_path("dummy.narrowPeak.bed.gz").unwrap();
+        assert!(RegionSet::try_from(file_path.to_str().unwrap()).is_ok());
+    }
+
+    #[test]
+    fn test_calculate_identifier() {
+        let file_path = get_test_path("dummy.narrowPeak.bed.gz").unwrap();
+        let region_set = RegionSet::try_from(file_path.to_str().unwrap()).unwrap();
+
+        assert_eq!("f0b2cf73383b53bd97ff525a0380f200", region_set.identifier());
+    }
+
+    #[test]
+    fn test_save_bed_gz() {
+        let file_path = get_test_path("dummy.narrowPeak.bed.gz").unwrap();
+        let region_set = RegionSet::try_from(file_path.to_str().unwrap()).unwrap();
+
+        let tempdir = tempfile::tempdir().unwrap();
+
+        let mut new_file_path = PathBuf::try_from(tempdir.into_path()).unwrap();
```

Comment:
useless conversion to the same type: `std::path::PathBuf`
consider removing `PathBuf::try_from()`
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
`#[warn(clippy::useless_conversion)]` on by default